### ISSUE TITLE
Button Group component

### ISSUE THREAD

**Issue by owestinTelus** [telus/tds-community#57]
## Problem Statement

* As a customer, I would like a clear and inviting method when filling forms 
* As a TELUS designer or developer, I would to use a repeatable pattern for making radio selection in forms, so that I can improve my speed to production

## Recommendation

* Use semantic form elements (radio group) styled as buttons to provide a clear visual presentation while communicating the recognizable pattern to assistive technology 
* Button Groups should be used as a replacement for a radio groups when the options are very short

## Reference
[Grouping controls - W3C Web Accessibility Initiative](https://www.w3.org/WAI/tutorials/forms/grouping/)

## Designs

![image](https://user-images.githubusercontent.com/12798751/41489093-a9d2b228-70bc-11e8-880a-16fb8a6634fd.png)

## Meta

* Willing to design solution: No
* Willing to develop solution: No
* Has workaround: No
* Do any similar components already exist: No
* Does this pattern exist already but isn’t a community component? 
  * Yes, on the family plan selection page on telus.com
* Which teams/applications do you think would use this improvement?
  * Mobility
* High impact: No

**Comment by ahmadnassri** [comment]
I'm not sure the problem statement is clear here?

**Comment by sketchidea** [comment]
@owestinTelus could you clarify the request as per @ahmadnassri 's comment

**Comment by theetrain** [comment]
I updated the original post for more clarity.

**Comment by theetrain** [comment]
This is now published! [@tds/core-button-group@1.0.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-button-group%401.0.0)
